enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+approvers:
+  - Miciah
+  - frobware
+  - gspence
+api-approvers:
+  - joel
+  - deads
+creation-date: 2024-03-06
+last-updated: 2024-03-22
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-1274"
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Set AWS EIP For NLB Ingress Controller
+
+## Summary
+
+This enhancement allows user to set AWS EIP for the NLB default or custom ingress controller.
+This is a feature request to enhance the IngressController API to be able to support static IPs during
+- Install time
+- Custom NLB ingress controller creation 
+- Reconfiguration of the custom and default router to NLB with EIP allocation.
+
+## Motivation
+
+### User Stories
+
+- As a cluster administrator, I want to provision EIPs and use them with an NLB IngressController.
+- As a cluster administrator, I want to ensure EIPs are used with NLB on default router at install time.
+- As a cluster administrator, I want to reconfigure default router to use EIPs.
+- As a cluster administrator of OpenShift on AWS (ROSA), I want to use static IPs (and therefore AWS Elastic IPs) so that 
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+- As a cluster administrator who has purchased a block of public IPv4 addresses, I want to use these IP addresses, so 
+  that I can avoid Amazon's new charge for public IP addresses.
+
+### Goals
+- Users are able to use EIP for a NLB Ingress Controller.
+- Users are able to create a NLB default Ingress Controller during install time with the EIPs specified during install.
+- The ELB of an existing IngressController should be able to be reconfigured to use specific EIPs by updating the IngressController CR and recreating the ELB with Load Balancer Type as `NLB`.
+
+### Non-Goals
+
+- Creation of EIPs in AWS.
+- Keep a track of changes in the EIP in the AWS environment.
+- Static IP usage with NLBs for API server.
+- Check the number of public subnets available and then compare number of subnets
+  with the number of EIPs the user provides.
+
+## Proposal
+
+EIP = AWS Elastic IP, NLB = AWS Network LoadBalancer 
+
+In this enhancement we would be adding API fields in the installer and the Ingress Controller CR
+to set the AWS EIP for the NLB Ingress Controller. The cluster ingress operator will then create 
+the ingress controller and get the EIP assigned to the service LoadBalancer with the annotation 
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+Note: `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` is a [kubernetes service annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/#service-beta-kubernetes-io-aws-load-balancer-eip-allocations).
+
+
+### Workflow Description
+
+Pre-requisites: One must [allocate Elastic IP address](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating)
```

Comment:
Why is this a prerequisite if we have a use case that references public IPv4 addresses not owned by AWS?

> As a cluster administrator who has purchased a block of public IPv4 addresses, I want to use these IP addresses, so that I can avoid Amazon's new charge for public IP addresses.

Reply:
I have updated it. We need either of them.